java/google/registry/rde/imports/XjcToDomainResourceConverter.java

```diff
@@ -181,14 +175,10 @@ public GracePeriod apply(XjcRgpStatusType gracePeriodStatus) {
   }
 
   /** Converts {@link XjcRdeDomain} to {@link DomainResource}. */
-  static DomainResource convertDomain(XjcRdeDomain domain) {
-    // First create history entry and autorenew billing event/poll message
-    // Autorenew billing event is required for creating AUTO_RENEW grace period
-    HistoryEntry historyEntry = createHistoryEntry(domain);
-    BillingEvent.Recurring autoRenewBillingEvent =
-        createAutoRenewBillingEvent(domain, historyEntry);
-    PollMessage.Autorenew pollMessage = createAutoRenewPollMessage(domain, historyEntry);
-    ofy().save().<ImmutableObject>entities(historyEntry, autoRenewBillingEvent, pollMessage);
+  static DomainResource convertDomain(
+      XjcRdeDomain domain,
+      BillingEvent.Recurring autoRenewBillingEvent,
+      PollMessage.Autorenew autoRenewPollMessage) {
     GracePeriodConverter gracePeriodConverter =
         new GracePeriodConverter(domain, Key.create(autoRenewBillingEvent));
```

Comment:
So I totally agree with your point here. The autorenew billing event/poll message were included here because the system required it in order to have an autorenew grace period. I think the cheapest approach here to actually handling grace period cancellations is to generate a report post-import of the TLD that flags them, and then issue refunds manually.